redaxo/src/core/lib/packages/null.php

```diff
@@ -42,39 +42,39 @@ public function getPackageId()
      */
     public function getPath($file = '')
     {
-        return null;
+        throw new rex_exception(sprintf('Calling %s on %s is not allowed', __FUNCTION__, __CLASS__));
```

Comment:
Hier war ich mir unsicher. 
null war problematisch, weil wir den Typ dann allgemein nullable machen müssten.
leeren string fände ich aber auch nicht so toll.

Vom Gefühl her denke ich, sollte man diese Methoden an null-packages eigentlich gar nicht aufrufen.
Finde es tendenziell problematisch, wenn sonst null oder leerer string als pfad weiter verwendet werden.

Reply:
Hmm wenn man hier exceptions wirft, macht man das null-objekt pattern nutzlos, oder?

Reply:
das pattern nutzen wir ja für die methoden `isAvailable` und `isInstalled` etc.
ich denke also dass man die methoden trennen kann, in solche die im null-object legitim aufgerufen werden können, und solche die dort nicht sinnvoll nutzbar sind.

Reply:
Mich stört sonst, dass es passieren kann, dass man mit 
`$addon->getPath().'/foo/bar'`
plötzlich den root-pfad `/foo/bar` erhalten kann.

Reply:
Dann sollten wir im klassen phpdoc klarstellen was im rahmen von null-objekt Supported ist/wozu es gedacht ist